core/modules/user/user.theme.inc

```diff
@@ -179,20 +188,37 @@ function theme_user_permission_description($variables) {
 function theme_user_admin_roles($variables) {
   $form = $variables['form'];
 
-  $header = array(t('Name'), t('Weight'), t('Operations'));
+  $header = array(
+    array('data' => t('Name'), 'class' => array('role-name')),
+    array('data' => t('Description'),'class' => array('role-description', 'priority-low')),
+    array('data' => t('Weight')),
+    array('data' => t('Operations'), 'class' => array('role-operations')),
+  );
   foreach (element_children($form['roles']) as $role_name) {
     $label = $form['roles'][$role_name]['#role']->label;
     if (in_array($role_name, array(BACKDROP_ANONYMOUS_ROLE, BACKDROP_AUTHENTICATED_ROLE))) {
       $label .= ' ' . t('(required)');
     }
 
     $row = array();
-    $row[] = theme('label_machine_name__role', array(
-      'label' => $label,
-      'machine_name' => $form['roles'][$role_name]['#role']->name,
-    ));
-    $row[] = backdrop_render($form['roles'][$role_name]['weight']);
-    $row[] = backdrop_render($form['roles'][$role_name]['operations']);
+    $row[] = array(
+      'data' => theme('label_machine_name__role', array(
+        'label' => $label,
+        'machine_name' => $form['roles'][$role_name]['#role']->name,
+      )),
+      'class' => array('role-name'),
+    );
+    $row[] = array(
+      'data' => filter_xss_admin(t($form['roles'][$role_name]['#role']->description)),
```

Comment:
We don't seem to be using `filter_xss_admin()`/`filter_xss()` for the label and machine name here. Should we? ...or should we be handling all this on form save instead? The only way for an attack vector here seems if someone manually edited the values of `user.role.*.json` config files.

Reply:
Looks like theme_label_machine_name() is what runs the check_plain() in this case, so we don't need to escape it here.